src/d/d_bg_s_acch.cpp

```diff
@@ -62,12 +114,144 @@ void dBgS_Acch::LineCheck(dBgS&) {
+    JUT_ASSERT(541, -1.0e32f < pm_pos->x && pm_pos->x < 1.0e32f && -1.0e32f < pm_pos->y && pm_pos->y < 1.0e32f && -1.0e32f < pm_pos->z && pm_pos->z < 1.0e32f);
+    
+    i_bgs.MoveBgCrrPos(m_gnd, ChkGroundHit(), pm_pos, pm_angle, pm_shape_angle);
+    
+    GroundCheckInit(i_bgs);
+    Init();
+    
+    f32 lowH_R = GetWallAllLowH_R();
+    cXyz* temp9 = pm_old_pos;
+    cXyz* temp10 = pm_pos;
+    f32 distXZ2 = temp9->abs2XZ(*temp10);
+    f32 distY = pm_old_pos->y - pm_pos->y;
+    f32 lowH = GetWallAllLowH();
+    field_0xb4 = pm_pos->y;
+    f32 oldY = pm_old_pos->y;
+    f32 temp7 = lowH + oldY;
+    f32 temp8 = field_0x98 + pm_pos->y;
+    
+    bool ranLineCheck = false;
+    OffLineCheckHit();
+    if (!ChkLineCheckNone() && !cM3d_IsZero(lowH_R)) {
+        if (distXZ2 > lowH_R*lowH_R || temp7 > temp8 || distY > field_0x98 || ChkLineCheck()) {
+            ranLineCheck = true;
+            LineCheck(i_bgs);
+        }
+    }
+    if (!(m_flags & WALL_NONE)) {
+        i_bgs.WallCorrect(this);
+    }
+    if (ChkWallHit() && ranLineCheck) {
+        LineCheck(i_bgs);
+    }
+    
+    m_roof_crr_height = 1000000000.0f;
+    if (!(m_flags & ROOF_NONE)) {
+        m_roof.SetExtChk(*(cBgS_Chk*)this);
+        ClrRoofHit();
+        cXyz roofPos = *pm_pos;
+        m_roof.SetPos(roofPos);
+        m_roof_y = i_bgs.RoofChk(&m_roof);
+        if (m_roof_y != 1000000000.0f && pm_pos->y + m_roof_height > m_roof_y) {
+            m_roof_crr_height = m_roof_y - m_roof_height;
+            SetRoofHit();
+        }
+    }
+    
+    if (!(m_flags & GRND_NONE)) {
+        ClrGroundFind();
+        GroundCheck(i_bgs);
+        GroundRoofProc(i_bgs);
+    } else if (m_roof_crr_height < pm_pos->y) {
+        pm_pos->y = m_roof_crr_height;
+    }
+    
+    if (!(m_flags & WATER_NONE)) {
+        ClrWaterHit();
+        ClrWaterIn();
+        m_wtr.SetHeight(-1000000000.0f);
+        
+        int room_no = i_bgs.GetRoomId(m_gnd);
+        if (m_ground_h != -1000000000.0f && 0 <= room_no && room_no < 64) {
+            JUT_ASSERT(693, 0 <= room_no && room_no < 64);
+            
+            dBgW* bgw = dComIfGp_roomControl_getBgW(room_no);
+            if (bgw) {
+                f32 top;
+                f32 under;
+                bgw->GetTopUnder(&top, &under);
+                
+                cXyz ground = *pm_pos;
+                ground.y = under;
+                m_wtr.Set(ground, top);
+                m_wtr.SetPassChkInfo(*this);
+                if (i_bgs.SplGrpChk(&m_wtr)) {
+                    SetWaterHit();
+                    if (m_wtr.GetHeight() > pm_pos->y) {
+                        SetWaterIn();
+                    }
+                    
+                    // JUT_ASSERT's (COND) == 0 check screws up the codegen here. Using !(COND) instead fixes it.
+                    // JUT_ASSERT(718, m_wtr.GetHeight() >= ground.y);
+                    // JUT_ASSERT(719, m_wtr.GetHeight() <= top);
```

Comment:
I was always a bit suspicious of that change, yeah, I'm curious what it would break to put it back.